addons/web/static/src/views/fields/parsers.js

```diff
@@ -133,6 +133,9 @@ export function parseInteger(value) {
             throw new InvalidNumberError(`"${value}" is not a correct number`);
         }
     }
+    if (parsed % 1 || parsed < -2147483648 || parsed > 2147483647) {
+        throw new InvalidNumberError(`"${value}" is not a correct number`);
```

Comment:
I feel like this message is not very clear as it does not specify that values are limited to integers, although I'm not sure how we could change it without getting too technical. Maybe `"${value}" is not a valid integer`? And the other message above at line 133 has the same issue, maybe it could be changed too?